src/components/containers/StudentDashboard/StudentWorkshopRegistration.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import UserContext from '../../../UserContext';
+import { ISession } from '../../../logic/domains/sessionDetails.domain';
+import getStudentWorkshops from '../../../logic/functions/getStudentWorkshops';
+import MaterialTable from 'material-table';
+import { RouteComponentProps } from 'react-router';
+import { Link, withRouter } from 'react-router-dom';
+
+type StudentWorkShopProps = RouteComponentProps<any> & {}
+
+let workshopData:any;
+
+const StudentWorkshopRegistration: React.FunctionComponent<StudentWorkShopProps> = (props) => {
+    const [sessions, setSessions] = React.useState<Array<ISession>>([]);
+    const context = React.useContext(UserContext);
+
+    React.useEffect(() => {
+        async function callGetWorkshops() {
+            try {
+                workshopData = await getStudentWorkshops(context.userDetails.studentId);
```

Comment:
don't do this. If we need to store state relevant to the component, use a state hook. 
See https://reactjs.org/docs/hooks-state.html
So what you could do (assuming your setState is called setWorkshopData) would be like:
 ```
let workshopData = await getStudentWorkshops(context.userDetails.studentId);
if (workshopData) {
  setWorkshopData(workshopData);
}
```
Also what is the type of workshop Data? You have `any`.